### PR TITLE
solution

### PR DESCRIPTION
**Pull request by Anton-Zhytonbaiev**
[DEMO LINK](https://Anton-Zhytonbaiev.github.io/react_people-table-basics/)

### CODE REVIEW

File: src/preperedPeople.ts [hunk 1]
```diff
@@ -0,0 +1,14 @@
+import { Person } from './types';
+
+export const preparePeopleArray = (peopleArray: Person[]) => {
```
Comment (oleh-holovnykh) [inline src/preperedPeople.ts:3]: 👍 nice solution

File: src/components/PeoplePage/PeoplePage.tsx [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+import { getPeople } from '../../api';
+import { Person } from '../../types';
+import { preparePeopleArray } from '../../preperedPeople';
+import { Loader } from '../Loader';
+import { PeopleTable } from '../PeopleTable/PeopleTable';
+
+export const PeoplePage: React.FC = () => {
+  const [people, setPeople] = useState<Person[] | []>([]);
+  const [isLoaderActive, setIsLoaderActive] = useState(true);
+  const [isLoadingError, setIisLoadingError] = useState(false);
```
Comment (oleh-holovnykh) [inline src/components/PeoplePage/PeoplePage.tsx:11]:
```suggestion
  const [isLoadingError, setIsLoadingError] = useState(false);
```

File: src/App.tsx [hunk 3]
```diff
@@ -12,159 +19,26 @@ export const App = () => (
...
-
-          <a
-            className="navbar-item has-background-grey-lighter"
-            href="#/people"
-          >
-            People
-          </a>
+          <PageNavLink to="/" text="Home" />
+          <PageNavLink to="/people" text="People" />
         </div>
```
Comment (oleh-holovnykh) [inline src/App.tsx:24]: 4. Add the `Navbar` with 2 links `Home` and `People`. Active one should have the `has-background-grey-lighter` class;

Create a separate component for navigation links

**Review by oleh-holovnykh — APPROVED** [review]
🔥 Well done! Take a look at the comments

File: src/preperedPeople.ts (hunk 1 above)
Comment (ArturErmolenko) [inline src/preperedPeople.ts:3]:
```suggestion
export const preparePeoples = (peopleArray: Person[]) => {
```

File: src/components/PersonLink/PersonLink.tsx [hunk 4]
```diff
@@ -0,0 +1,25 @@
+import classNames from 'classnames';
+import React from 'react';
+import { Link } from 'react-router-dom';
+import { Person } from '../../types';
+
+type Props = {
+  person: Person,
+};
+
+export const PersonLink: React.FC<Props> = ({ person }) => {
```
Comment (ArturErmolenko) [inline src/components/PersonLink/PersonLink.tsx:10]: memo

File: src/components/PersonItem/PersonItem.tsx [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+      </td>
+
+      <td>{sex}</td>
+      <td>{born}</td>
+      <td>{died}</td>
+      <td>
+        {(
+          mother
+            ? <PersonLink person={mother} />
+            : motherName || '-'
```
Comment (ArturErmolenko) [inline src/components/PersonItem/PersonItem.tsx:42]: ternary operator should return one value depending on condition, don't make another condition in the ternary operator

File: src/components/PersonItem/PersonItem.tsx [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+        )}
+      </td>
+      <td>
+        {(
+          father
+            ? <PersonLink person={father} />
+            : fatherName || '-'
```
Comment (ArturErmolenko) [inline src/components/PersonItem/PersonItem.tsx:49]: same

File: src/components/PeoplePage/PeoplePage.tsx [hunk 7]
```diff
@@ -0,0 +1,63 @@
...
+      {isLoadingError && (
+        <p
+          data-cy="peopleLoadingError"
+          className="has-text-danger"
+        >
+          Something went wrong
+        </p>
+      )}
+
+      {people.length > 0 && (
```
Comment (ArturErmolenko) [inline src/components/PeoplePage/PeoplePage.tsx:50]:
```suggestion
      {!!people.length && (
```

File: src/components/PeoplePage/PeoplePage.tsx [hunk 8]
```diff
@@ -0,0 +1,63 @@
+import { useEffect, useMemo, useState } from 'react';
+import { getPeople } from '../../api';
+import { Person } from '../../types';
+import { preparePeopleArray } from '../../preperedPeople';
+import { Loader } from '../Loader';
+import { PeopleTable } from '../PeopleTable/PeopleTable';
+
+export const PeoplePage: React.FC = () => {
```
Comment (ArturErmolenko) [inline src/components/PeoplePage/PeoplePage.tsx:8]: memo

**Review by ArturErmolenko — CHANGES_REQUESTED** [review]

**Review by ArturErmolenko — APPROVED** [review]
Great 🚀

**Pull request closed without merging**